### ISSUE TITLE
selectedFiles[I].name always contains 'cached_<uuid>' instead of the file name on Android 10

### ISSUE THREAD

**Issue by raalkml** [Waboodoo/HTTP-Shortcuts#215]
I use a before-request scripting section in a simple http shortcut. The section supposed to pass the name of the picked file to the remote machine for saving the file. 

 ```js
 setVariable("my_file_name", selectedFiles[0].name);
```

I'm trying to use this selectedFiles to send files using "Share" or "Send To" in apps on my phone, but for some reason the name of the file is never the name of the original file. It is rather something like "shared_<some-uuid>". The file size and type get sensible values, just not the name. 

Do you know if this an Android 10 feature (hiding filenames for whatever reason) which I can deactivate? The storage permission is enabled, but it does not make any difference if I disable it.

The Android on the phone (Motorola One Zoom) is a Motorola branded Android 10.

 I tried to use selectedFiles on an Android 7 (Lineage on a Nexus7) system, where it does work.

_Originally posted by @raalkml in https://github.com/Waboodoo/HTTP-Shortcuts/issues/207#issuecomment-774557549_

**Comment by raalkml** [comment]
I don't know if this of any use, but I tried Termux (it can also receive files with Send/Share) and the file name there seem to work: Termux receiving code lets you edit the name before working on it. Looking at the [Termux code](https://github.com/termux/termux-app/blob/4139bf94243d1f593ee668a633d5dc75c298d39d/app/src/main/java/com/termux/filepicker/TermuxFileReceiverActivity.java#L68) they seem to use something called "Intent.EXTRA_TITLE" as a [fallback](https://github.com/termux/termux-app/blob/master/app/src/main/java/com/termux/filepicker/TermuxFileReceiverActivity.java#L94) if something else, called DISPLAY_NAME (which HTTP Shortcuts also use) did not work:

```java
handleContentUri(sharedUri, intent.getStringExtra(Intent.EXTRA_TITLE));
...
void handleContentUri(final Uri uri, String subjectFromIntent) {
        try {
            String attachmentFileName = null;

            String[] projection = new String[]{OpenableColumns.DISPLAY_NAME};
            try (Cursor c = getContentResolver().query(uri, projection, null, null, null)) {
                if (c != null && c.moveToFirst()) {
                    final int fileNameColumnId = c.getColumnIndex(OpenableColumns.DISPLAY_NAME);
                    if (fileNameColumnId >= 0) attachmentFileName = c.getString(fileNameColumnId);
                }
            }

            if (attachmentFileName == null) attachmentFileName = subjectFromIntent;

            InputStream in = getContentResolver().openInputStream(uri);
            promptNameAndSave(in, attachmentFileName);
        } catch (Exception e) {
            showErrorDialogAndQuit("Unable to handle shared content:\n\n" + e.getMessage());
            Log.e("termux", "handleContentUri(uri=" + uri + ") failed", e);
        }
    }
```


```

**Comment by Waboodoo** [comment]
Thanks for reporting this bug in such a detailed way. I will fix this issue for the next release.

**Comment by raalkml** [comment]
Seems to be fixed in play store 2.1.0. Cool :+1:
